### PR TITLE
Added new Tuya v2 Integration

### PR DESCRIPTION
**Pull request by zlinoliver**
## Proposed change

This PR is for the release of the Tuya v2 integration: https://github.com/home-assistant/core/tree/tuya-v2

Tuya Smart always strive to provide the best development experience possible and keep updating our Home Assistant integration capabilities. Since the old version of Tuya Home Assistant integration supports limited categories and control commands, now we bring you our new integration with extended support for more categories and commands.

Our new integration is built on Tuya’s standard [Open APIs](https://developer.tuya.com/en/docs/cloud) and tackles the limits of categories and accessible control commands. We deliver more development-friendly features including virtual device debugging, official authorization, and technical supports, which will help you efficiently work on development with the Tuya ecosystem.

The existing [Tuya Integration](https://www.home-assistant.io/integrations/tuya/) is no longer maintained by Tuya officially, and will be deprecated. The estimated deprecation date is 2021.12.31. The new integration will be officially maintained by the Tuya Developer Team, for more information of the migration, please check this documentation [How to Migrate the Tuya Home Assistant Integration from the Old Version](https://developer.tuya.com/en/docs/iot/migrate-from-an-older-version?id=Kamee9wtbd00b)

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/56820
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/tuya.markdown [hunk 1]
```diff
@@ -25,79 +24,115 @@ ha_platforms:
...
+
+#### Create a project
+
+1. Log in to the [Tuya IoT Platform](https://iot.tuya.com/).
+2. In the left navigation bar, click **Cloud** > **Projects**.
+3. On the page that appears, click **Create**.
+4. In the **Create Project** dialog box, configure **Project Name**, **Description**, **Industry**, and **Availability Zone**. In the **Development Method** field, select **Smart Home PaaS** from the dropdown list.
+	> **Note:** Select all available zones.
+	
+![](../images/integrations/tuya/image_001.png)
```
Comment (balloob) [inline source/_integrations/tuya.markdown]:
```suggestion
![](/images/integrations/tuya/image_001.png)
```

File: source/_integrations/tuya.markdown [hunk 2]
```diff
@@ -25,79 +24,115 @@ ha_platforms:
...
+	
+5. Click **Create** to continue project configuration.
+6. In **Configuration Wizard**, select **Device status notification** API.
+   > **Note:** **Smart Home PaaS** API has been selected as default.
+   
+![](../images/integrations/tuya/image_002.png)
```
Comment (balloob) [inline source/_integrations/tuya.markdown]:
```suggestion
![](/images/integrations/tuya/image_002.png)
```

File: source/_integrations/tuya.markdown [hunk 3]
```diff
@@ -25,79 +24,115 @@ ha_platforms:
...
+
+7. Click **Authorize**.
+
+#### Get authorization key
 
-During configuration, be careful to select the [country code](https://www.countrycode.org/) and the platform corresponding to those used by you in the app. Once configuration flow is completed, the devices configured in your app will be automatically discovered. Additionally, if your username or e-mail address isn't accepted, please try using your phone number (minus the country code) as your username.
+Click the created project to enter the **Project Overview** page and get the **Authorization Key** used to make API calls.
 
-## Integration Options
+![](../images/integrations/tuya/image_003.png)
```
Comment (balloob) [inline source/_integrations/tuya.markdown]:
```suggestion
![](/images/integrations/tuya/image_003.png)
```

File: source/_integrations/tuya.markdown [hunk 4]
```diff
@@ -25,79 +24,115 @@ ha_platforms:
...
-- **Discovery device polling interval** (default=605): define the interval between 2 consecutive calls to the `API discovery method`, which is used to get the status for all registered devices with a single call. If you set interval value too low, Tuya API will return errors, so it is suggested to use the default value until
-you know that it is possible to use lower values.
+**1.** Navigate to the **Devices** tab.
 
-- **Query device polling interval** (default=120): this option is available only if you have devices that can use the `API query method`. 
-It defines the interval between 2 consecutive calls to the `API query method`, that is used to get the status for a specific device. 
-This method is always used when it is available only one device that can use it. If you set interval value too low, Tuya API will return errors 
-so it is suggested to use the default value until you know that is possible to use lower values.
+**2.** Click **Link Tuya App Account** > **Add App Account**.
+![](../images/integrations/tuya/image_004.png)
```
Comment (balloob) [inline source/_integrations/tuya.markdown]:
```suggestion

![](/images/integrations/tuya/image_004.png)
```

File: source/_integrations/tuya.markdown [hunk 5]
```diff
@@ -25,79 +24,115 @@ ha_platforms:
...
 
-- **Device that will use the query method**: this option is available only if you have devices that can use the `API query method`. 
-Because it is not possible to make multiple calls to the `API query method`. If you have more than one device that can use it you can choose which one will use. This will give a better status refresh for this specific device.
+**3.** Scan the QR code that appears using the **Tuya Smart** app or **Smart Life** app.
+![](../images/integrations/tuya/image_005.png)
```
Comment (balloob) [inline source/_integrations/tuya.markdown]:
```suggestion

![](/images/integrations/tuya/image_005.png)
```

File: source/_integrations/tuya.markdown [hunk 6]
```diff
@@ -25,79 +24,115 @@ ha_platforms:
...
 
-- **Device to configure (multi-select list)**: this option is available only if you have a `light` or `climate` device. Selecting a device to 
-configure to options page related to the device will be opened. You can also select more than one devices to configure them simultaneously, 
-but all selected devices must be of the same type.
+**4.** Click **Confirm** on your **Tuya Smart** or **Smart Life** app.
 
-### Light Options
+**5.** Navigate to the **Device List** tab. You can find the devices that have been added through your **Tuya Smart** or **Smart Life** account.   
+![](../images/integrations/tuya/image_006.png)
```
Comment (balloob) [inline source/_integrations/tuya.markdown]:
```suggestion

![](/images/integrations/tuya/image_006.png)
```

File: source/_integrations/tuya.markdown [hunk 7]
```diff
@@ -25,79 +24,115 @@ ha_platforms:
...
 
-- **Force color support**: when checked force `color support` for devices that do not report this feature.
+</details>
```
Comment (balloob) [inline source/_integrations/tuya.markdown]:
```suggestion
```

File: source/_integrations/tuya.markdown [hunk 8]
```diff
@@ -25,79 +24,115 @@ ha_platforms:
...
 
-- **Max color temperature reported**: set the maximum `color temperature` value reported by the light.
+You can also get help from [Set up Home Assistant Development Environment on Raspberry Pi](https://developer.tuya.com/en/demo/setuphomeassistantdevenv/) in Tuya Developer Demo Center.
 
-### Climate Options
+#### Install the Tuya integration
 
-- **Temperature unit**: change the `temperature unit` used internally by the devices.
+In **Configuration** > **Integrations** > **ADD INTEGRATION**, search for the keyword **Tuya** and select it for installation.
+![](../images/integrations/tuya/image_007.png)
```
Comment (balloob) [inline source/_integrations/tuya.markdown]: This can all be removed. People are in the HA docs.

File: source/_integrations/tuya.markdown [hunk 9]
```diff
@@ -25,79 +24,115 @@ ha_platforms:
...
-- **Max target temperature**: set the maximum allowed `target temperature` for the entity.
+| Field | Description |
+| ------- | -------- |
+| Access ID and Access Secret| Go to your cloud project on [Tuya IoT Platform](https://iot.tuya.com/). Find the **Access ID** and **Access Secret** on the **Project Overview** tab.|
+| Mobile App | Must select the one you used to link devices on the Tuya IoT Platform. |
+| Country Code | The country you select on logging in to Tuya Smart or Smart Life app.|
+| Account | Tuya Smart or Smart Life app account. |
+| Password | The password of your app account. |
+> **Note**: The app mentioned in the table must be the one you used to link devices on the Tuya IoT Platform.
+</details>
```
Comment (balloob) [inline source/_integrations/tuya.markdown]:
```suggestion

<p class='note'>The app mentioned in the table must be the one you used to link devices on the Tuya IoT Platform.</p>
```

File: source/_integrations/tuya.markdown [hunk 10]
```diff
@@ -25,79 +24,115 @@ ha_platforms:
...
+
+You have created an account for [Tuya Smart or Smart Life app](https://developer.tuya.com/en/docs/iot/tuya-smart-app-smart-life-app-advantages?id=K989rqa49rluq#title-1-Download).
+
+#### Create a project
+
+1. Log in to the [Tuya IoT Platform](https://iot.tuya.com/).
+2. In the left navigation bar, click **Cloud** > **Projects**.
+3. On the page that appears, click **Create**.
+4. In the **Create Project** dialog box, configure **Project Name**, **Description**, **Industry**, and **Availability Zone**. In the **Development Method** field, select **Smart Home PaaS** from the dropdown list.
+	> **Note:** Select all available zones.
```
Comment (balloob) [inline source/_integrations/tuya.markdown]:
```suggestion
4. In the **Create Project** dialog box, configure **Project Name**, **Description**, **Industry**, and **Availability Zone**. In the **Development Method** field, select **Smart Home PaaS** from the dropdown list. Select all available zones.
```

File: source/_integrations/tuya.markdown [hunk 11]
```diff
@@ -25,79 +24,115 @@ ha_platforms:
 ha_dhcp: true
 ---
 
-The `tuya` integration is the main integration to integrate [Tuya Smart](https://www.tuya.com) related platforms, except the Zigbee and Bluetooth(BLE) hub. This includes devices linked with the Tuya, Smart Life, and Jinvoo Smart apps. You will need your account information (username, password and account country code) from one of these apps to discover and control devices which related to your account.
+The Tuya Home Assistant integration is developed for controlling **Powered by Tuya (PBT)** devices, which is officially maintained by the Tuya Developer Team.
```
Comment (balloob) [inline source/_integrations/tuya.markdown]:
```suggestion
The Tuya integration allows integrating all Powered by Tuya devices you have added to the Tuya Smart and Tuya Smart Life apps. This integration is officially maintained by Tuya.
```

File: source/_integrations/tuya.markdown [hunk 12]
```diff
@@ -25,79 +24,115 @@ ha_platforms:
...
 
-**Important**: Not all Tuya devices are supported by the `tuya API` used by this integration. For more details refer to [TuyaHA Library](https://github.com/PaulAnnekov/tuyaha).
+It applies to smart devices that you have added to the Tuya Smart or Smart Life app.
```
Comment (balloob) [inline source/_integrations/tuya.markdown]:
```suggestion
```

File: source/_integrations/tuya.markdown [hunk 13]
```diff
@@ -25,79 +24,115 @@ ha_platforms:
...
-- **Light** - The platform supports most kinds of Tuya light.
-- **Scene** - The device state in frontend panel will not change immediately after you activate a scene.
-- **Switch** - The platform supports switch and socket.
+## Supported platforms
+- **Climate**: Supports air conditioner, heater, thermostat.
+- **Cover**: Supports curtain, curtain switch.
+- **Fan**: Supports fan, air purifier, humidifier, dehumidifier.
+- **Light**: Supports light, light strip, ambient light, light string, humidifier's light, Ceiling light.
+- **Scene**: Supports smart scene.
+- **Switch**: Supports switch, socket, power strip, smart kettle, breaker, pet water feeder, air purifier, diffuser.
```
Comment (balloob) [inline source/_integrations/tuya.markdown]:
```suggestion
- **Climate**: air conditioner, heater, thermostat.
- **Cover**: curtain, curtain switch.
- **Fan**: fan, air purifier, humidifier, dehumidifier.
- **Light**: light, light strip, ambient light, light string, humidifier's light, Ceiling light.
- **Scene**: smart scene.
- **Switch**: switch, socket, power strip, smart kettle, breaker, pet water feeder, air purifier, diffuser.
```

File: source/_integrations/tuya.markdown (hunk 1 above)
Comment (zlinoliver) [reply source/_integrations/tuya.markdown]: fixed

File: source/_integrations/tuya.markdown (hunk 2 above)
Comment (zlinoliver) [reply source/_integrations/tuya.markdown]: fixed

File: source/_integrations/tuya.markdown (hunk 3 above)
Comment (zlinoliver) [reply source/_integrations/tuya.markdown]: fixed

File: source/_integrations/tuya.markdown (hunk 4 above)
Comment (zlinoliver) [reply source/_integrations/tuya.markdown]: fixed

File: source/_integrations/tuya.markdown (hunk 5 above)
Comment (zlinoliver) [reply source/_integrations/tuya.markdown]: fixed

File: source/_integrations/tuya.markdown (hunk 6 above)
Comment (zlinoliver) [reply source/_integrations/tuya.markdown]: fixed

File: source/_integrations/tuya.markdown (hunk 7 above)
Comment (zlinoliver) [reply source/_integrations/tuya.markdown]: removed.

File: source/_integrations/tuya.markdown (hunk 8 above)
Comment (zlinoliver) [reply source/_integrations/tuya.markdown]: OK, we will keep the "Install the Tuya integration" section.

File: source/_integrations/tuya.markdown (hunk 9 above)
Comment (zlinoliver) [reply source/_integrations/tuya.markdown]: fixed.

File: source/_integrations/tuya.markdown (hunk 10 above)
Comment (zlinoliver) [reply source/_integrations/tuya.markdown]: fixed.

File: source/_integrations/tuya.markdown (hunk 11 above)
Comment (zlinoliver) [reply source/_integrations/tuya.markdown]: fixed.

File: source/_integrations/tuya.markdown (hunk 12 above)
Comment (zlinoliver) [reply source/_integrations/tuya.markdown]: fixed

File: source/_integrations/tuya.markdown (hunk 13 above)
Comment (zlinoliver) [reply source/_integrations/tuya.markdown]: fixed.

**Review by balloob — APPROVED** [review]

**Pull request merged**